cli/configssh.go

```diff
@@ -269,6 +316,26 @@ func configSSH() *cobra.Command {
 			if err != nil {
 				return xerrors.Errorf("fetch workspace configs failed: %w", err)
 			}
+
+			coderdConfig, err := client.SSHConfiguration(ctx)
+			if err != nil {
+				// If the error is 404, this deployment does not support
+				// this endpoint yet. Do not error, just assume defaults.
+				// TODO: Remove this in 2 months (May 2023). Just return the error
+				// 	and remove this 404 check.
+				var sdkErr *codersdk.Error
+				if !xerrors.As(err, &sdkErr) {
+					// not an SDK error, return the original error.
+					return xerrors.Errorf("fetch coderd config failed: %w", err)
+				}
+
+				if sdkErr.StatusCode() != http.StatusNotFound {
+					// Not a 404, return the original error.
+					return xerrors.Errorf("fetch coderd config failed: %w", err)
+				}
```

Comment:
This is just to support new coder cli against an older coderd. Can drop it after some time.

Reply:
Meh, I don't think we should do this. Our install script supports installing by version, so it's really easy to upgrade/downgrade.

Reply:
If we don't care, I can drop it. I think it is mostly something developers will hit.

Reply:
What error will users see when they do this on an old client?

Reply:
As long as it's clear to the they need to upgrade to resolve it, I don't think we need to have backwards compatibility.

Reply:
Let's be sure to label this as a breaking change though

Reply:
Can I just keep this then @kylecarbs? It makes things backwards compatible without docs needed. I included a drop date to remove the code.

It's just a convince to handle any edge cases, and it's not expensive or complex.